### PR TITLE
Rename `kafka-connect` to `kafka-connector`

### LINKED ISSUE #149: Rename `kafka-connect` to `kafka-connector`
Rename:

- [x]  [`kafka_connect.py`](https://github.com/bakdata/kpops/blob/main/kpops/components/base_components/kafka_connect.py)
- [x] rename [`type`](https://github.com/bakdata/kpops/blob/main/kpops/components/base_components/kafka_connect.py#L36)
- [x] rename `schema_type`, run `gen_schema`
- [x] update all docs, do https://github.com/bakdata/kpops-examples/issues/9

 `kafka-connect` --> `kafka-connector`

### PR DESCRIPTION
**Pull request by sujuka99**
closes #149

### CODE REVIEW

**Comment by sujuka99** [comment]
NOTE: You may notice that the example defaults for atm-fraud are not updated. This is because they are pulled from the main branch. As soon as this PR is merged, the docs will become correct.

Ideally the atm-fraud example should be moved to [kpops-examples](https://github.com/bakdata/kpops-examples)

File: docs/docs/resources/pipeline-defaults/defaults-kafka-connect.yaml
Comment (disrupted) [inline docs/docs/resources/pipeline-defaults/defaults-kafka-connect.yaml:1]: please also rename the file

File: docs/docs/user/getting-started/quick-start.md [hunk 1]
```diff
@@ -48,7 +48,7 @@ redis bitnami/redis
 
 #### Port forwarding
 
-Before we deploy the pipeline, we need to forward the ports of `kafka-rest-proxy` and `kafka-connect`.
+Before we deploy the pipeline, we need to forward the ports of `kafka-rest-proxy` and `kafka-connector`.
```
Comment (disrupted) [inline docs/docs/user/getting-started/quick-start.md:51]: this should stay, as it is referring to the Kafka Connect server

File: docs/docs/user/examples/atm-fraud-pipeline.md
Comment (disrupted) [inline docs/docs/user/examples/atm-fraud-pipeline.md:1]: those changes are not correct, should stay

**Review by disrupted — CHANGES_REQUESTED** [review]

File: docs/docs/resources/pipeline-defaults/defaults-kafka-connect.yaml
Comment (sujuka99) [reply docs/docs/resources/pipeline-defaults/defaults-kafka-connect.yaml:1]: done

File: docs/docs/user/examples/atm-fraud-pipeline.md
Comment (sujuka99) [reply docs/docs/user/examples/atm-fraud-pipeline.md:1]: my replacing was a bit sloppy it seems, fixed

File: docs/docs/user/getting-started/quick-start.md (hunk 1 above)
Comment (sujuka99) [reply docs/docs/user/getting-started/quick-start.md:51]: done

**Review by raminqaf — APPROVED** [review]

**Review by disrupted — APPROVED** [review]
LGTM

**Pull request merged**